From 5053cea087fd54ce6d978852aae5163cebcac8fc Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Thu, 29 Nov 2012 17:22:17 +0200
Subject: [PORT FROM MAIN] Telephony:Disable excessive MO SMS check-"eng" build

BZ: 61977

Currently, there is a per-application based
SMS control, which checks the excessive
outgoing SMS usage. Upon limit reached,
an Alert dialog will be presented to the user
for confirmation of sending the SMS. Due to this
dialog, ACS test is blocked.

In engineering builds, this check will limit
the automated testing.

This patch disables the excessive outgoing
SMS check on "eng" build type only

Change-Id: I3136976708500fe6c5a34b8a5a168b63491c0ea1
Orig-Change-Id: I903ba63beeecebc52e0b275a82b73944dea5da9b
Category: device enablement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../android/internal/telephony/SMSDispatcher.java  |   11 +++++++----
 1 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/java/com/android/internal/telephony/SMSDispatcher.java b/src/java/com/android/internal/telephony/SMSDispatcher.java
index 0258ff3..c71b250 100644
--- a/src/java/com/android/internal/telephony/SMSDispatcher.java
+++ b/src/java/com/android/internal/telephony/SMSDispatcher.java
@@ -696,10 +696,13 @@ public abstract class SMSDispatcher extends Handler {
         // sending app is approved to send to short codes. Otherwise, a message is sent to our
         // handler with the SmsTracker to request user confirmation before sending.
         if (checkDestination(tracker)) {
-            // check for excessive outgoing SMS usage by this app
-            if (!mUsageMonitor.check(appInfo.packageName, SINGLE_PART_SMS)) {
-                sendMessage(obtainMessage(EVENT_SEND_LIMIT_REACHED_CONFIRMATION, tracker));
-                return;
+            String buildType = SystemProperties.get("ro.build.type", null);
+            if (buildType == null || !buildType.equals("eng")) {
+                // check for excessive outgoing SMS usage by this app
+                if (!mUsageMonitor.check(appInfo.packageName, SINGLE_PART_SMS)) {
+                    sendMessage(obtainMessage(EVENT_SEND_LIMIT_REACHED_CONFIRMATION, tracker));
+                    return;
+                }
             }
 
             int ss = mPhone.getServiceState().getState();
-- 
1.7.4.1

